### ISSUE TITLE
Return the original status code to the caller on service invocation. Error based retry configuration

### ISSUE THREAD

**Issue by mhozhda-agiliway** [dapr/dapr#3826]
## In what area(s)?

Dapr service invocation (version 1.4.3) used via dot.net sdk (version 1.4)
Dotnet version: dotnet 5.0

## What version of Dapr?
1.4.3

## Expected Behavior

After calling other service endpoint via dapr service invocation I expect to get the actual status code of the response.

## Actual Behavior

The service returns 404 (NotFoundObjectResult) though dapr performs a few retries and then returns 500 status code
to the caller app. Also it would be great to have the possibility to configure retries base on status code returned.
E.g in my app 404 is expected response when there is no data and I don't want it to be retried.

## Steps to Reproduce the Problem

Just create two simple apps using dotnet 5.0 and dapr (1.4.3) and try to use service invocation when
one app returns 404 NotFound status code. But the caller after some retries gets 500 as a result

## Release Note

RELEASE NOTE: Return original status code on service invocation. Add the possibility to configure retries base on error type

**Comment by yaron2** [comment]
Triaged to 1.6

**Comment by yaron2** [comment]
Reopened

**Comment by artursouza** [comment]
@yaron2, I am moving this to 1.7

**Comment by yaron2** [comment]
@mhozhda-agiliway I took a look at this and from what I can see Dapr does the right thing:

I ran two apps, A and B.

App B exposed a single endpoint: `/api/orders`.

I invoked App B from App A using Dapr with the following url `v1.0/invoke/appA/method/api/something` and received `404 Not Found`.
That's the expected response, as Dapr was able to discovery App B, route the message to it, and returned the 404 message that it received from the target web server.

I then invoked a non existent app id (`v1.0/invoke/appC/method/api/something`), and after several retries received a `500` error response with the following message: `fail to invoke, id: appC, err: couldn't find service: appC`.

This is expected because Dapr failed the discovery phase, and so this is not a response from the target service, but an error from the Dapr runtime.
